lib/sync/api-syncRecords.js

```diff
@@ -19,7 +19,7 @@ function listLocalDatasetClientData(datasetClient, cb) {
     }
     //no one sync loop has completed yet, return null
     if (!datasetClientsWithRecords.syncCompleted) {
-      return cb();
+      return cb(null, null);
```

Comment:
It actually does. See the test I have added in this PR. It will fix the other error we saw in the ticket. The problem is that this is being used as part of `async.waterfall`, and it nothing is returned (`undefined`), then the `async` module will not pass it to the next function. But if it's `null` value, `async` will still pass it as the first parameter to the next function.